.github/workflows/pull_request.yml

```diff
@@ -107,10 +148,50 @@ jobs:
         run: ./scripts/run_validator.sh
       - name: Running schema tests
         run: ./scripts/validate_test_schemas.sh
+  lint-javascript:
```

Comment:
Any reason to not use existing `python-dependencies` and `node-dependencies`

Reply:
This also isn't saving us much time, as you need python and node deps for the JS one also, since generate pages is python. I think the lint can just be a separate job, to unit test, but I would do python and js lint in same job as 2 steps (can be run in parallel).

Reply:
> Any reason to not use existing `python-dependencies` and `node-dependencies`

I deleted virtualenv install steps everywhere and cache pipenv using the newer syntax. I don't think there's much more we can improve here.